src/braket/jobs/hybrid_job.py

```diff
@@ -217,13 +221,49 @@ def job_wrapper(*args, **kwargs) -> Callable:
                         job_args[key] = value
 
                 job = _create_job(job_args, local)
+                shutil.rmtree(inner_source_temp_dir)
             return job
 
         return job_wrapper
 
     return _hybrid_job
 
 
+def persist_inner_function_source(entry_point: callable) -> tuple[str, str]:
+    """Persist the mapping between the name and the source code for each inner function inside the
+    entry point, as a dictionary in a json file.
+
+    Args:
+        entry_point (callable): The job decorated function.
+
+    Returns:
+        tuple[str, str]: Temporary directory and path of the json file.
+    """
+    inner_source = _get_inner_function_source(entry_point)
+    temp_dir = tempfile.mkdtemp()
+    inner_source_file_path = f'{temp_dir}/inner_function_source.json'
+    with open(inner_source_file_path, "w") as inner_source_file:
+        json.dump(inner_source, inner_source_file)
+    return temp_dir, inner_source_file_path
+
+
+def _get_inner_function_source(outer_function: callable) -> dict[str, str]:
+    """Returns a dictionary that maps the function name to source code for all inner functions
+    inside the job decorated function.
+
+    Args:
+        outer_function (callable): The outer function.
+
+    Returns:
+        dict[str, str]: Mapping between name and source code of each inner functions.
+    """
+    inner_function_source = {}
+    for const in outer_function.__code__.co_consts:
```

Comment:
1. Does this work recursively? e.g. what if my job-decorated function calls another function, and the `@aq.main` is defined there?
2. Does this work for `@aq.main`-decorated functions that are defined outside the scope of the job-decorated function (or imported from another module)?

Reply:
Both do not work.
For (1), I think if `my_func` is in  job-decorated function, there is a way to work. 
For (2), function defined outside is tricky. Need to play with it to see.

Reply:
Recursive function is now supported. The solution for function defined outside of job decorator remains elusive. It likely requires a remake of how job decorator works, removing dependency of cloudpickle.